### PR TITLE
feat: add index classes

### PR DESCRIPTION
**Pull request by averikitsch**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/langchain-google-cloud-sql-pg-python/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: src/langchain_google_cloud_sql_pg/indexes.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+import enum
+from typing import List, Optional
+
+
+class DistanceStrategy(str, enum.Enum):
+    """Enumerator of the Distance strategies."""
+
+    EUCLIDEAN = "l2"
+    COSINE = "cosine"
```
Comment (kurtisvg) [inline src/langchain_google_cloud_sql_pg/indexes.py:23]:
```suggestion
    COSINE_DISTANCE = "cosine"
```

File: src/langchain_google_cloud_sql_pg/indexes.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import enum
+from typing import List, Optional
+
+
+class DistanceStrategy(str, enum.Enum):
```
Comment (kurtisvg) [inline src/langchain_google_cloud_sql_pg/indexes.py:19]:
```suggestion
class DistanceStrategy(enum.Enum):
```

File: src/langchain_google_cloud_sql_pg/indexes.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+    EUCLIDEAN = "l2"
+    COSINE = "cosine"
+    INNER_PRODUCT = "inner"
+
+
+DEFAULT_DISTANCE_STRATEGY = DistanceStrategy.COSINE
+
+
+class BruteForce:
```
Comment (kurtisvg) [inline src/langchain_google_cloud_sql_pg/indexes.py:30]: Can we have an abstract base class between all of these?

File: src/langchain_google_cloud_sql_pg/indexes.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    ):
+        self.distance_strategy = distance_strategy
+
+
+class HNSWIndex:
+    index_type = "hnsw"
+
+    def __init__(
+        self,
+        name: str = "langchainhnsw",
```
Comment (kurtisvg) [inline src/langchain_google_cloud_sql_pg/indexes.py:45]: I think we decided we should have a common default name here

File: src/langchain_google_cloud_sql_pg/indexes.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+class DistanceStrategy(str, enum.Enum):
+    """Enumerator of the Distance strategies."""
+
+    EUCLIDEAN = "l2"
+    COSINE = "cosine"
+    INNER_PRODUCT = "inner"
+
+
+DEFAULT_DISTANCE_STRATEGY = DistanceStrategy.COSINE
```
Comment (kurtisvg) [inline src/langchain_google_cloud_sql_pg/indexes.py:27]: Why cosign?

File: src/langchain_google_cloud_sql_pg/indexes.py [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    index_type = "knn"
+
+    def __init__(
+        self,
+        distance_strategy: DistanceStrategy = DEFAULT_DISTANCE_STRATEGY,
+    ):
+        self.distance_strategy = distance_strategy
+
+
+class HNSWIndex:
```
Comment (kurtisvg) [inline src/langchain_google_cloud_sql_pg/indexes.py:40]: Any reason not to use `@dataclass`?

File: src/langchain_google_cloud_sql_pg/indexes.py [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+        self.name = name
+        self.lists = lists
+        self.partial_indexes = partial_indexes
+        self.distance_strategy = distance_strategy
+        self.query_options = self.QueryOptions()
+
+    def index_options(self) -> str:
+        return f"(lists = {self.lists})"
+
+    class QueryOptions:
```
Comment (kurtisvg) [inline src/langchain_google_cloud_sql_pg/indexes.py:85]: How do we plan on using this?

**Review by kurtisvg — CHANGES_REQUESTED** [review]

File: src/langchain_google_cloud_sql_pg/indexes.py (hunk 5 above)
Comment (averikitsch) [reply src/langchain_google_cloud_sql_pg/indexes.py]: https://platform.openai.com/docs/guides/embeddings/which-distance-function-should-i-use

**Pull request closed without merging**